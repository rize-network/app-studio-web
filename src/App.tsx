import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
