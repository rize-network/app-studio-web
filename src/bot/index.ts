import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
