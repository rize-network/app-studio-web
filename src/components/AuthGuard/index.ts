export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
