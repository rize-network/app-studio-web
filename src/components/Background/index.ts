export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
} from './Background/Background.type';
