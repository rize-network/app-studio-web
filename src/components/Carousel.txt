'use client';
import React, { useEffect, useRef, useState, createContext } from 'react';
import { Horizontal, Vertical } from './Layout';
import { IconButtons } from './Button/examples';
import { Button } from './Button/Button';
import { RightArrowSvg } from 'src/components/Svg/index';

// import { IconLeft, IconRight } from '@app-studio/web/icons';

interface CarouselProps {
  items: JSX.Element[];
  initialScroll?: number;
}

export const CarouselContext = createContext({
  onCardClose: (index: number) => {},
  currentIndex: 0,
});

export const Carousel = ({ items, initialScroll = 0 }: CarouselProps) => {
  const carouselRef = useRef<HTMLDivElement>(null);
  const [canScrollLeft, setCanScrollLeft] = useState(false);
  const [canScrollRight, setCanScrollRight] = useState(true);

  useEffect(() => {
    if (carouselRef.current) {
      carouselRef.current.scrollLeft = initialScroll;
      checkScrollability();
    }
  }, [initialScroll]);

  const checkScrollability = () => {
    if (carouselRef.current) {
      const { scrollLeft, scrollWidth, clientWidth } = carouselRef.current;
      setCanScrollLeft(scrollLeft > 0);
      setCanScrollRight(scrollLeft < scrollWidth - clientWidth);
    }
  };

  const scrollLeft = () =>
    carouselRef.current?.scrollBy({ left: -300, behavior: 'smooth' });
  const scrollRight = () =>
    carouselRef.current?.scrollBy({ left: 300, behavior: 'smooth' });

  return (
    <CarouselContext.Provider
      value={{ onCardClose: () => {}, currentIndex: 0 }}
    >
      <Horizontal
        gap={2}
        alignItems="center"
        width="100%"
        paddingVertical={20}
        position="relative"
      >
        <Button
          onClick={scrollLeft}
          icon={<RightArrowSvg />}
          disabled={!canScrollLeft}
          shape="rounded"
        />
        <Horizontal
          ref={carouselRef}
          onScroll={checkScrollability}
          gap={4}
          overflowX="auto"
          paddingX={10}
        >
          {items.map((item, index) => (
            <Vertical key={index} minWidth="300px">
              {item}
            </Vertical>
          ))}
        </Horizontal>
        <Button
          onClick={scrollRight}
          icon={<RightArrowSvg />}
          disabled={!canScrollRight}
          shape="rounded"
        />
      </Horizontal>
    </CarouselContext.Provider>
  );
};
