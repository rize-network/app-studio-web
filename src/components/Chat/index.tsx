/**
 * App Studio Components for AI Applications
 *
 * This module exports a collection of components specifically designed
 * for building AI-powered applications.
 */

// Core components
export * from './ChatInterface/ChatInterface';
export * from './MessageBubble/MessageBubble';
export * from './PromptInput/PromptInput';
export * from './ModelSelector/ModelSelector';
export * from './AIResponseDisplay/AIResponseDisplay';
export * from './LoadingIndicator/LoadingIndicator';

// Supporting components
export * from './ToolSelector/ToolSelector';
export * from './SettingsPanel/SettingsPanel';
export * from './FileUploader/FileUploader';
export * from './CodeBlock/CodeBlock';
