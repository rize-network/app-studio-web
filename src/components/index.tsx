export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
