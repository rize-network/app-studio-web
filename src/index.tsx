import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
