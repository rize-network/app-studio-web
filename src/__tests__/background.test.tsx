import React from 'react';
import renderer from 'react-test-renderer';
import { Background } from 'src/components';

import { cleanup, render, screen } from '@testing-library/react';

afterEach(() => {
  cleanup();
});

describe('Background Component', () => {
  test('renders Aurora background without crashing', () => {
    render(
      <Background.Aurora data-testid="aurora-background">
        <div>Aurora Content</div>
      </Background.Aurora>
    );
    const auroraElement = screen.getByTestId('aurora-background');
    expect(auroraElement).toBeInTheDocument();
    expect(auroraElement).toHaveTextContent('Aurora Content');
  });

  test('renders Meteors background without crashing', () => {
    render(<Background.Meteors data-testid="meteors-background" number={5} />);
    const meteorsElement = screen.getByTestId('meteors-background');
    expect(meteorsElement).toBeInTheDocument();
  });

  test('renders Wall background without crashing', () => {
    render(
      <Background.Wall 
        data-testid="wall-background" 
        rows={5} 
        cols={5} 
        squareSize={20} 
      />
    );
    const wallElement = screen.getByTestId('wall-background');
    expect(wallElement).toBeInTheDocument();
  });

  test('Aurora background with showRadialGradient prop', () => {
    render(
      <Background.Aurora 
        data-testid="aurora-gradient" 
        showRadialGradient={true}
      >
        Content
      </Background.Aurora>
    );
    const auroraElement = screen.getByTestId('aurora-gradient');
    expect(auroraElement).toBeInTheDocument();
  });

  test('Meteors background with custom number of meteors', () => {
    render(<Background.Meteors data-testid="meteors-custom" number={10} />);
    const meteorsElement = screen.getByTestId('meteors-custom');
    expect(meteorsElement).toBeInTheDocument();
  });

  test('Wall background with custom dimensions', () => {
    render(
      <Background.Wall 
        data-testid="wall-custom" 
        rows={8} 
        cols={6} 
        squareSize={30} 
      />
    );
    const wallElement = screen.getByTestId('wall-custom');
    expect(wallElement).toBeInTheDocument();
  });
});

describe('Background Snapshots', () => {
  test('Aurora background to match snapshot', () => {
    const tree = renderer
      .create(
        <Background.Aurora showRadialGradient={true}>
          Aurora Content
        </Background.Aurora>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Meteors background to match snapshot', () => {
    const tree = renderer
      .create(<Background.Meteors number={3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Wall background to match snapshot', () => {
    const tree = renderer
      .create(<Background.Wall rows={3} cols={3} squareSize={20} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
