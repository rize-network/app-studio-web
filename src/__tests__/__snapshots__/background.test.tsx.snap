// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background Snapshots Aurora background to match snapshot 1`] = `
<div
  className="pn-relative dy-flex fdn-column ht-100vh ais-center jct-center"
>
  <div
    style={
      Object {
        "animation": "aurora 4s ease infinite",
        "backgroundImage": "
      repeating-linear-gradient(100deg, rgb(59, 130, 246), rgb(196, 181, 253), rgb(147, 197, 253), rgb(221, 214, 254), rgb(96, 165, 250)),
      repeating-linear-gradient(100deg, rgba(255,255,255,1) 0%, rgba(255,255,255,1) 7%, rgba(255,255,255,0) 10%, rgba(255,255,255,0) 12%, rgba(255,255,255,1) 16%)
    ",
        "backgroundPosition": "50% 50%, 50% 50%",
        "backgroundSize": "300% 200%",
        "filter": "blur(10px)",
        "inset": "-10px",
        "maskImage": "radial-gradient(ellipse at 100% 0%, black 10%, transparent 70%)",
        "opacity": 0.5,
        "pointerEvents": "none",
        "position": "absolute",
      }
    }
  />
  Aurora Content
  <style>
    
          @keyframes aurora {
            0% { background-position: 0% 50%; }
            50% { background-position: 100% 50%; }
            100% { background-position: 0% 50%; }
          }
        
  </style>
</div>
`;

exports[`Background Snapshots Meteors background to match snapshot 1`] = `
<div
  className="wh-300px ht-300px wht-300 bcr- pn-relative"
>
  <div
    className="wh-2px ht-2px animation-name-animation-15 animation-duration-1500ms animation-timing-function-linear animation-delay-360p7849773885239ms animation-iteration-count-infinite animation-direction-normal animation-fill-mode-forwards animation-play-state-running pn-absolute tp-0px lt--63px wh-2px ht-2px brs-9999px bcr- bsw-0-0-0-1px-rgba255-255-255-0p1 tm-rotate215deg"
  >
    <div
      className="wh-1px ht-1px pn-absolute tp--1px lt--1px wh-1px ht-1px brs-9999px bcr- bsw-0-10px-0-1px-rgba255-255-255-0p1"
    />
  </div>
  <div
    className="wh-2px ht-2px animation-name-animation-3 animation-duration-2500ms animation-timing-function-linear animation-delay-446p10529346519803ms animation-iteration-count-infinite animation-direction-normal animation-fill-mode-forwards animation-play-state-running pn-absolute tp-0px lt--37px wh-2px ht-2px brs-9999px bcr- bsw-0-0-0-1px-rgba255-255-255-0p1 tm-rotate215deg"
  >
    <div
      className="wh-1px ht-1px pn-absolute tp--1px lt--1px wh-1px ht-1px brs-9999px bcr- bsw-0-10px-0-1px-rgba255-255-255-0p1"
    />
  </div>
  <div
    className="wh-2px ht-2px animation-name-animation-6 animation-duration-1s animation-timing-function-linear animation-delay-513p0741086063239ms animation-iteration-count-infinite animation-direction-normal animation-fill-mode-forwards animation-play-state-running pn-absolute tp-0px lt-87px wh-2px ht-2px brs-9999px bcr- bsw-0-0-0-1px-rgba255-255-255-0p1 tm-rotate215deg"
  >
    <div
      className="wh-1px ht-1px pn-absolute tp--1px lt--1px wh-1px ht-1px brs-9999px bcr- bsw-0-10px-0-1px-rgba255-255-255-0p1"
    />
  </div>
</div>
`;

exports[`Background Snapshots Wall background to match snapshot 1`] = `
<div
  className="wh-300px ht-300px pg-16px wht-300 zix-0 tm-skewX-48deg-skewY14deg-scale0p675-rotate0deg-translateZ0"
>
  <div
    className="dy-flex fdn-row"
  >
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
  </div>
  <div
    className="dy-flex fdn-row"
  >
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
  </div>
  <div
    className="dy-flex fdn-row"
  >
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
    <div
      className="ht-6p666666666666667px wh-20px bsee-solid bcrr- bwh-0p5px bcr---hover"
    />
  </div>
</div>
`;
